sources/web/datalab/polymer/modules/drive-file-manager/drive-file-manager.ts

```diff
@@ -52,6 +52,15 @@ class DriveFileManager extends BaseFileManager {
     return this._fromUpstreamFile(upstreamFile);
   }
 
+  public newFileNameError(fileName: string): string | null {
+    // Must match _getWhitelistFilePredicates()
+    if (fileName.indexOf('.ipynb') > -1 || fileName.indexOf('.txt') > -1) {
```

Comment:
Shouldn't this be "endsWith" of something instead of indexOf?

Also, I feel like we should have a function or something in one place that controls the list of acceptable file extensions.

Reply:
I was trying to make it match the search criteria used in _getWhitelistFilePredicates, but I guess we can be more strict here and require the .txt to be at the end of the name. Done.
Re having a function in one place, at present the filename filtering is on a per-file-manager basis; for example, the jupyter file manager shows all files. We can add a master type list later once we figure out what it should control.